tests/fs-test.rs

```diff
@@ -39,73 +61,187 @@ fn new_file_test() {
 	check_result(&res);
 }
 
+/// This is expected to fail.
 #[test]
-fn uhyvefilemap_test() {
+#[serial]
+fn create_write_unmapped_nonpresent_file() {
 	env_logger::try_init().ok();
-	let output_path = PathBuf::from("foo.txt");
-	remove_file_if_exists(&output_path);
-	let bin_path = build_hermit_bin("create_file");
+	let mut host_path = get_fs_fixture_path();
+	host_path.push("ignore_everything_here");
+	host_path.push("file_to_write.txt");
+	remove_file_if_exists(&host_path);
+
+	let uhyvefilemap_params = [format!(
+		"{}:{}",
+		host_path.to_str().unwrap(),
+		"/root/dir/wrong.txt"
+	)]
+	.to_vec();
 
-	let mut params = Params {
+	let params = Params {
 		cpu_count: 2.try_into().unwrap(),
 		memory_size: Byte::from_u64_with_unit(64, Unit::MiB)
 			.unwrap()
 			.try_into()
 			.unwrap(),
-		file_mapping: vec!["/root/foo.txt:wrong.txt".to_string()],
+		file_mapping: uhyvefilemap_params,
 		output: Output::Buffer,
 		stats: true,
 		..Default::default()
 	};
 
 	// The file should not exist on the host OS.
-	let mut vm = UhyveVm::new(bin_path.clone(), params.clone()).unwrap();
-	let mut res = vm.run(None);
+	let bin_path = build_hermit_bin("create_file");
+	let vm = UhyveVm::new(bin_path.clone(), params.clone()).unwrap();
+	let res = vm.run(None);
 	check_result(&res);
-	assert!(!output_path.exists());
+	assert!(!host_path.exists());
+}
+
+#[test]
+#[serial]
+fn create_write_mapped_nonpresent_file() {
+	env_logger::try_init().ok();
+	let mut host_path = get_fs_fixture_path();
+	host_path.push("ignore_everything_here");
+	host_path.push("foo.txt");
+	remove_file_if_exists(&host_path);
+
+	let uhyvefilemap_params = [format!(
+		"{}:{}",
+		host_path.to_str().unwrap(),
+		"/root/foo.txt"
+	)]
+	.to_vec();
+
+	let params = Params {
+		cpu_count: 2.try_into().unwrap(),
+		memory_size: Byte::from_u64_with_unit(64, Unit::MiB)
+			.unwrap()
+			.try_into()
+			.unwrap(),
+		file_mapping: uhyvefilemap_params,
+		output: Output::Buffer,
+		stats: true,
+		..Default::default()
+	};
+
+	let bin_path = build_hermit_bin("create_file");
+	let vm = UhyveVm::new(bin_path.clone(), params.clone()).unwrap();
+	let res = vm.run(None);
 
-	params.file_mapping = vec!["foo.txt:/root/foo.txt".to_string()];
-	vm = UhyveVm::new(bin_path, params).unwrap();
-	res = vm.run(None);
 	check_result(&res);
-	verify_file_equals(&output_path, "Hello, world!");
+	verify_file_equals(&host_path, "Hello, world!");
 }
 
+/// This might break because of a misconfiguration in Landlock.
```

Comment:
What does the author of these lines want to tell us? :thinking:

Reply:
The reason why the function might fail *because* of a regression. I can rephrase this.